Source/Laser/LaserProfilesImpl/LaserProfileHarris.cpp

```diff
@@ -0,0 +1,79 @@
+#include <LaserProfiles.H>
+
+#include <WarpX_Complex.H>
+#include <WarpXConst.H>
+
+using namespace amrex;
+using namespace WarpXLaserProfiles;
+
+void
+HarrisLaserProfile::init (
+    const amrex::ParmParse& ppl,
+    const amrex::ParmParse& ppc,
+    CommonLaserParameters params)
+{
+    // Parse the properties of the Harris profile
+    ppl.get("profile_waist", m_params.waist);
+    ppl.get("profile_duration", m_params.duration);
+    ppl.get("profile_focal_distance", m_params.focal_distance);
+    //Copy common params
+    m_common_params = params;
+}
+
+/* \brief compute field amplitude for a Harris laser function, at particles' position
+ *
+ * Both Xp and Yp are given in laser plane coordinate.
+ * For each particle with position Xp and Yp, this routine computes the
+ * amplitude of the laser electric field, stored in array amplitude.
+ *
+ * \param np: number of laser particles
+ * \param Xp: pointer to first component of positions of laser particles
+ * \param Yp: pointer to second component of positions of laser particles
+ * \param t: Current physical time
+ * \param amplitude: pointer to array of field amplitude.
+ */
+void
+HarrisLaserProfile::fill_amplitude (
+    const int np, Real const * AMREX_RESTRICT const Xp, Real const * AMREX_RESTRICT const Yp,
+    Real t, Real * AMREX_RESTRICT const amplitude)
+{
+    // This function uses the Harris function as the temporal profile of the pulse
+    const Real omega0 =
+        2._rt*MathConst::pi*PhysConst::c/m_common_params.wavelength;
+    const Real zR = MathConst::pi * m_params.waist*m_params.waist
+        / m_common_params.wavelength;
+    const Real wz = m_params.waist *
+        std::sqrt(1._rt + m_params.focal_distance*m_params.focal_distance/zR*zR);
```

Comment:
this looks like a bug to me:
```suggestion
        std::sqrt(1._rt + m_params.focal_distance*m_params.focal_distance/(zR*zR));
```

Reply:
Yes, it's definitively a bug! Thanks for having spotted it! While refactoring the class I just copied the content of the original function, which had the bug!

Reply:
Nice! :)
That's how I feel now: https://www.youtube.com/watch?v=S65ZzffW-Cw